pkg/controller/jobpost/jobpost.go

```diff
@@ -0,0 +1,166 @@
+  posts, err := jobpost.FetchAllJobPost(base.Db.Postgresql)
+  if err != nil {
+    rd := utility.BuildErrorResponse(http.StatusInternalServerError, "error", "Failed to fetch job posts", err, nil)
+    c.JSON(http.StatusInternalServerError, rd)
+    return
+  }
+
+  base.Logger.Info("posts fetched successfully")
+  rd := utility.BuildSuccessResponse(http.StatusOK, "", posts)
+  c.JSON(http.StatusOK, rd)
+}
+
+func (base *Controller) FetchJobPostById(c *gin.Context) {
+	jobPostID := c.Param("id")
+
+  if _, err := uuid.Parse(jobPostID); err != nil {
+		rd := utility.BuildErrorResponse(http.StatusBadRequest, "error", "Invalid job post ID", err, nil)
+		c.JSON(http.StatusBadRequest, rd)
+		return
+	}
+	post, err := jobpost.FetchJobPostById(base.Db.Postgresql, jobPostID)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			rd := utility.BuildErrorResponse(http.StatusNotFound, "error", "Job post not found", err, nil)
+			c.JSON(http.StatusNotFound, rd)
+			return
+		}
+		rd := utility.BuildErrorResponse(http.StatusInternalServerError, "error", "Failed to fetch job post", err, nil)
+		c.JSON(http.StatusInternalServerError, rd)
+		return
+	}
+
+	base.Logger.Info("post fetched successfully")
+	rd := utility.BuildSuccessResponse(http.StatusOK, "", post)
+	c.JSON(http.StatusOK, rd)
+}
+
+func (base *Controller) UpdateJobPostById(c *gin.Context) {
+	jobPostID := c.Param("id")
+
+  if _, err := uuid.Parse(jobPostID); err != nil {
+		rd := utility.BuildErrorResponse(http.StatusBadRequest, "error", "Invalid job post ID", err, nil)
+		c.JSON(http.StatusBadRequest, rd)
+		return
+	}
+
+
+	var body struct {
+		Title       *string  `json:"title"` 
+		Description *string  `json:"description"`
+		Location    *string  `json:"location"`
+		Salary      *float64 `json:"salary"`      
+		JobType     *string  `json:"job_type"`
+		CompanyName *string  `json:"company_name"`
+	}
+
+	if err := c.BindJSON(&body); err != nil {
+		rd := utility.BuildErrorResponse(http.StatusBadRequest, "error", "Invalid request body", err, nil)
+		c.JSON(http.StatusBadRequest, rd)
+		return
+	}
+
+	updates := map[string]interface{}{}
+	if body.Title != nil {
+		updates["title"] = *body.Title
+	}
+	if body.Description != nil {
+		updates["description"] = *body.Description
+	}
+	if body.Location != nil {
+		updates["location"] = *body.Location
+	}
+	if body.Salary != nil {
+		updates["salary"] = *body.Salary
+	}
+	if body.JobType != nil {
+		updates["job_type"] = *body.JobType
+	}
+	if body.CompanyName != nil {
+		updates["company_name"] = *body.CompanyName
+	}
+
+	post, err := jobpost.UpdateJobPostById(base.Db.Postgresql, jobPostID, updates)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			rd := utility.BuildErrorResponse(http.StatusNotFound, "error", "Job post not found", err, nil)
+			c.JSON(http.StatusNotFound, rd)
+			return
+		}
+		rd := utility.BuildErrorResponse(http.StatusInternalServerError, "error", "Failed to update job post", err, nil)
```

Comment:
use service for business logic
Look at what others have done please

Reply:
Ok, thanks for the review. I will effect changes now.